src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DocumentGeneratedOutputTracker.cs

```diff
@@ -50,6 +50,13 @@ public Task<RazorCodeDocument> GetGeneratedOutputInitializationTask(ProjectSnaps
                     if (_task == null)
                     {
                         _task = GetGeneratedOutputInitializationTaskCore(project, document);
+
+                        // We wait until after we've set the task so any triggers from SetOutput can acquire the 
+                        // the generated output from the document snapshot.
+                        if (document is DefaultDocumentSnapshot defaultDocument)
+                        {
+                            defaultDocument.State.HostDocument.GeneratedCodeContainer.SetOutput(document);
+                        }
```

Comment:
Would there be a problem just changing the parameter type? `DocumentSnapshot` isn't an abstraction, it just has an internal + external surface area.

Reply:
Are you suggesting having SetOutput take a `DefaultDocumentSnapshot`? If that's the case I'd rather just add a `HostDocument` property to `DocumentSnapshot`

Reply:
yes, and no. HostDocument is outside the facade.

Reply:
Interesting. I guess if we wanted to set output for a different document type we could always overload.

Reply:
There won't be another document type.